File: c++/QEasyWidgets/Components/SpinBox.cpp
#include "SpinBox.h"

#include "../Common/StyleSheet.h"


namespace QEW {

SpinBoxBase::SpinBoxBase(QWidget *parent)
    : QSpinBox(parent)
{
    init();
}

void SpinBoxBase::init()
{
    setFocusPolicy(Qt::StrongFocus);
    StyleSheetBase::apply(this, StyleSheetBase::SpinBox);
}

void SpinBoxBase::wheelEvent(QWheelEvent *event)
{
    event->ignore();
}

void SpinBoxBase::setBorderless(bool borderless)
{
    setProperty("isBorderless", borderless);
}

void SpinBoxBase::setTransparent(bool transparent)
{
    setProperty("isTransparent", transparent);
}

void SpinBoxBase::clearDefaultStyleSheet()
{
    // TODO: Implement deregistration if StyleSheetBase supports it
}

// DoubleSpinBoxBase implementation
DoubleSpinBoxBase::DoubleSpinBoxBase(QWidget *parent)
    : QDoubleSpinBox(parent)
{
    init();
}

void DoubleSpinBoxBase::init()
{
    setFocusPolicy(Qt::StrongFocus);
    StyleSheetBase::apply(this, StyleSheetBase::SpinBox);
}

void DoubleSpinBoxBase::wheelEvent(QWheelEvent *event)
{
    event->ignore();
}

void DoubleSpinBoxBase::setBorderless(bool borderless)
{
    setProperty("isBorderless", borderless);
}

void DoubleSpinBoxBase::setTransparent(bool transparent)
{
    setProperty("isTransparent", transparent);
}

void DoubleSpinBoxBase::clearDefaultStyleSheet()
{
    // TODO: Implement deregistration if StyleSheetBase supports it
}

} // namespace QEW

File: python/QEasyWidgets/QEasyWidgets/Components/SpinBox.py
from typing import Optional, overload
from PySide6.QtGui import *
from PySide6.QtCore import *
from PySide6.QtWidgets import *

from ..Common.StyleSheet import *
from ..Common.QFunctions import *
from .Menu import MenuBase

##############################################################################################################################

class SpinBoxBase(QSpinBox):
    """
    Base class for spinBox components
    """

    def __init__(self, parent: Optional[QWidget] = None):
        super().__init__(parent)

        self.setFocusPolicy(Qt.StrongFocus)

        StyleSheetBase.SpinBox.apply(self)

    def wheelEvent(self, event: QWheelEvent) -> None:
        event.ignore()

    def setBorderless(self, borderless: bool) -> None:
        self.setProperty("isBorderless", borderless)

    def setTransparent(self, transparent: bool) -> None:
        self.setProperty("isTransparent", transparent)

    def clearDefaultStyleSheet(self) -> None:
        StyleSheetBase.SpinBox.deregistrate(self)

##############################################################################################################################

class DoubleSpinBoxBase(QDoubleSpinBox):
    """
    Base class for doubleSpinBox components
    """

    def __init__(self, parent: Optional[QWidget] = None):
        super().__init__(parent)

        self.setFocusPolicy(Qt.StrongFocus)

        StyleSheetBase.SpinBox.apply(self)

    def wheelEvent(self, event: QWheelEvent) -> None:
        event.ignore()

    def setBorderless(self, borderless: bool) -> None:
        self.setProperty("isBorderless", borderless)

    def setTransparent(self, transparent: bool) -> None:
        self.setProperty("isTransparent", transparent)

    def clearDefaultStyleSheet(self) -> None:
        StyleSheetBase.SpinBox.deregistrate(self)

##############################################################################################################################